### PR TITLE
Raise `Disconnect` on `send()` when client disconnected

### PR DESCRIPTION
**Pull request by Kludex**
- Comply with https://github.com/django/asgiref/pull/431.

### CODE REVIEW

File: uvicorn/protocols/utils.py [hunk 1]
```diff
@@ -6,6 +6,10 @@
 from uvicorn._types import WWWScope
 
 
+class Disconnected(IOError):
```
Comment (adriangb) [inline uvicorn/protocols/utils.py]: Maybe ClientDisconnected?

**Review by adriangb — COMMENTED** [review]
What about http?

**Comment by Kludex** [comment]
> What about http?

I don't know yet, but shouldn't be a blocker for this PR.

**Comment by adriangb** [comment]
Ok just wanted to know it will be handled in a future PR

**Review by adriangb — APPROVED** [review]

**Comment by Kludex** [comment]
> Ok just wanted to know it will be handled in a future PR

I have personal plans to work on this on the next days, but I don't want to promise. 👀

**Comment by adriangb** [comment]
No need to promise. I just wanted to make sure you're not proposing that it doesn't get implemented by anyone at any time (i.e. this PR is all we're going to do).

File: uvicorn/protocols/utils.py (hunk 1 above)
Comment (Kludex) [reply uvicorn/protocols/utils.py]: done

File: uvicorn/protocols/utils.py [hunk 2]
```diff
@@ -6,6 +6,10 @@
 from uvicorn._types import WWWScope
 
 
+class ClientDisconnected(IOError):
```
Comment (graingert) [inline uvicorn/protocols/utils.py:9]: IOError is an alias for OSError on python 3

Comment (adriangb) [reply uvicorn/protocols/utils.py:9]: Ouch I think we put `IOError` in the spec...

**Pull request merged**